python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -599,18 +292,97 @@ class RandomForestClassifier(Base):
+
+        cdef size_t params_t = <size_t> self.rf_forest
+        cdef  RandomForestMetaData[float, int] *rf_forest = \
+            <RandomForestMetaData[float, int]*>params_t
+
+        cdef size_t params_t64 = <size_t> self.rf_forest64
+        cdef  RandomForestMetaData[double, int] *rf_forest64 = \
+            <RandomForestMetaData[double, int]*>params_t64
 
-        self._impl = RandomForest_impl(n_estimators, max_depth, self.handle,
-                                       max_features, n_bins,
-                                       split_algo, split_criterion,
-                                       min_rows_per_node,
-                                       bootstrap, bootstrap_features,
-                                       type_model, verbose,
-                                       rows_sample, max_leaves,
-                                       quantile_per_tree,
-                                       gdf_datatype)
+        state['verbose'] = self.verbose
+
+        if self.dtype == np.float32:
+            print(" dtype is float32")
+            state["rf_params"] = rf_forest.rf_params
+            del state["rf_forest"]
+        else:
+            print(" dtype is float64")
+            state["rf_params64"] = rf_forest64.rf_params
+            del state["rf_forest64"]
+
+        return state
+
+    def __del__(self):
+        cdef RandomForestMetaData[float, int]* rf_forest = \
+            <RandomForestMetaData[float, int]*><size_t> self.rf_forest
+        cdef RandomForestMetaData[double, int]* rf_forest64 = \
+            <RandomForestMetaData[double, int]*><size_t> self.rf_forest64
+        free(rf_forest)
+        free(rf_forest64)
+
+    def __setstate__(self, state):
+
+        super(RandomForestClassifier, self).__init__(handle=None,
+                                                     verbose=state['verbose'])
+        cdef  RandomForestMetaData[float, int] *rf_forest = \
+            new RandomForestMetaData[float, int]()
+        cdef  RandomForestMetaData[double, int] *rf_forest64 = \
+            new RandomForestMetaData[double, int]()
+
+        if self.dtype == np.float32:
+            print(" dtype is float32 in set state")
+            rf_forest.rf_params = state["rf_params"]
+            state["rf_forest"] = <size_t>rf_forest
+        else:
+            print(" dtype is float64 in set state")
+            rf_forest64.rf_params = state["rf_params64"]
+            state["rf_forest64"] = <size_t>rf_forest64
+
+        self.__dict__.update(state)
+
+    def _get_type(self):
+        if self.criterion == 0:
+            return GINI
+        elif self.criterion == 1:
+            return ENTROPY
+        elif self.criterion == 2:
+            return MSE
+        elif self.criterion == 3:
+            return MAE
+        else:
+            return CRITERION_END
+
+    def _get_max_feat_val(self):
+        if type(self.max_features) == int:
+            max_feature_val = self.max_features/self.n_cols
```

Comment:
you can remove the `max_val_feature_val` variable and just return the calculation directly